### PR TITLE
Item Price update in Price List

### PR DESCRIPTION
**Pull request by KanchanChauhan**
https://github.com/frappe/erpnext/issues/5251 fixed

### CODE REVIEW

File: erpnext/stock/get_item_details.py [hunk 1]
```diff
@@ -237,20 +237,30 @@ def insert_item_price(args):
 	if frappe.db.get_value("Price List", args.price_list, "currency") == args.currency \
 		and cint(frappe.db.get_single_value("Stock Settings", "auto_insert_price_list_rate_if_missing")):
 		if frappe.has_permission("Item Price", "write"):
-
 			price_list_rate = args.rate / args.conversion_factor \
 				if args.get("conversion_factor") else args.rate
-
+			if not price_list_rate: return
+		
 			item_price = frappe.get_doc({
```
Comment (nabinhait) [inline erpnext/stock/get_item_details.py]: This should be in else block

File: erpnext/stock/get_item_details.py [hunk 2]
```diff
@@ -237,20 +237,30 @@ def insert_item_price(args):
...
 				"price_list": args.price_list,
 				"item_code": args.item_code,
 				"currency": args.currency,
 				"price_list_rate": price_list_rate
 			})
-			item_price.insert()
-			frappe.msgprint(_("Item Price added for {0} in Price List {1}").format(args.item_code,
-				args.price_list))
+			
+			name = frappe.db.get_value('Item Price', {'item_code': args.item_code, 'price_list': args.price_list}, 'name')
```
Comment (nabinhait) [inline erpnext/stock/get_item_details.py]: Also consider currency while checking for existing item price record

File: erpnext/stock/get_item_details.py (hunk 1 above)
Comment (KanchanChauhan) [inline erpnext/stock/get_item_details.py]: I realized, this line was not needed at all. Removed it.

File: erpnext/stock/get_item_details.py (hunk 2 above)
Comment (KanchanChauhan) [inline erpnext/stock/get_item_details.py]: Added currency to filter.

**Pull request merged**